azure-core/azure/core/paging.py

```diff
@@ -0,0 +1,113 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+import sys
+try:
+    from collections.abc import Iterator
+    xrange = range
+except ImportError:
+    from collections import Iterator
+
+from typing import Dict, Any, List, Callable, Optional, TYPE_CHECKING  # pylint: disable=unused-import
+
+from msrest.serialization import Deserializer
```

Comment:
Could we remove this dependency and pass in a function rather than classes?
The Deserilazer is a `callable` and is only used for one operation - so I think we should pass through the callable into the constructor and use it directly.
This would require a small modification to autorest to instantiate the Deserializer before passing it in.

Reply:
I made changes for this in https://github.com/Azure/azure-sdk-for-python/pull/4873/commits/170ac22b9f5c3cfbc1d1cc9a101a5a3008eb4ff7 

Perhaps another option would be to inject the result of `Deserializer(classes)` then `classes would not need to be passed in?

Reply:
Ah - sorry @bryevdv - yes that was my meaning - that we pass in the result of Deserializer(classes) and remove classes as a parameter. That way the deserializer we pass in could be any arbitrary callable and this class could be used by non-autorest-generated code.
This should also be a very simple change for autorest.

Reply:
@annatisch I made this change in https://github.com/Azure/azure-sdk-for-python/pull/4873/commits/f3d7412ba875b38c8d6e4bbb1dd9285617f860ba

Reply:
also https://github.com/Azure/azure-sdk-for-python/pull/4873/commits/16ef88e8198633bbf89b55362da39020472ce403 (typings)